src/compiler/crystal/codegen/crystal_llvm_builder.cr

```diff
@@ -41,8 +41,14 @@ module Crystal
       value
     end
 
-    def printf(format, args = [] of LLVM::Value)
-      call @printf, [global_string_pointer(format)] + args
+    def printf(format, args = [] of LLVM::Value, catch_pad = nil)
```

Comment:
call to printf do really need to be handled with the funclet? Up to know the call was never replaced by an invoke, so it was assumed to never raise AFAIK.

Reply:
*every* call inside a catch pad needs a funclet. This is not about call/invoke and `@rescue_block`, it's about `@catch_pad` which is a totally different concern and semantic. This is about telling LLVM which method calls are inside the "catch block", and it applies to *every single method call*. `@rescue_block` is about codegenning `invoke` instructions, and it only matters for calls that raise.

Reply:
Ok. I also missed this was inside the builder, outside the codegen where the handling of the operand bundle is already handled.